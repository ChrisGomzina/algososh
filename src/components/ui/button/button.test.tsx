import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from "./button";

describe("Корректная отрисовка", () => {
  it("кнопки с текстом", () => {
    const button = renderer.create(<Button text="text" />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("кнопки без текста", () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("заблокированной кнопки", () => {
    const button = renderer.create(<Button disabled={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("кнопки с индикацией загрузки", () => {
    const button = renderer.create(<Button isLoader={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("вызова колбека при клике на кнопку", () => {
    window.alert = jest.fn();
    const onClick = () => alert("Кнопка нажата");
    render(<Button onClick={onClick} text={"text"} />);
    const button = screen.getByText("text");
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith("Кнопка нажата");
  });
});