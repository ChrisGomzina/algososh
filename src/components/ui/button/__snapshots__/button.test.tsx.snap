// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Корректная отрисовка заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Корректная отрисовка кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Корректная отрисовка кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Корректная отрисовка кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;
